src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreGetResponseTests.cs

```diff
@@ -18,107 +18,89 @@ namespace Dapr.PluggableComponents.Components.StateStore;
 public sealed class StateStoreGetResponseTests
 {
     [Fact]
-    public void ToGetResponseNullTest()
+    public void ToGetResponseNull()
     {
-        var grpcResponse = StateStoreGetResponse.ToGetResponse(null);
+        var response = StateStoreGetResponse.ToGetResponse(null);
 
-        Assert.Equal(String.Empty, grpcResponse.ContentType);
-        Assert.Equal(new byte[] { }, grpcResponse.Data);
-        Assert.Null(grpcResponse.Etag);
-        Assert.Empty(grpcResponse.Metadata);
+        Assert.Equal(String.Empty, response.ContentType);
+        Assert.Empty(response.Data);
+        Assert.Null(response.Etag);
+        Assert.Empty(response.Metadata);
     }
 
-    [Theory]
-    [InlineData(null, "")]
-    [InlineData("", "")]
-    [InlineData("application/json", "application/json")]
-    public void ToGetResponseContentTypeTests(string? contentType, string expectedContentType)
-    {
-        var response = new StateStoreGetResponse
-        {
-            ContentType = contentType
-        };
-
-        var grpcResponse = StateStoreGetResponse.ToGetResponse(response);
-
-        Assert.Equal(expectedContentType, grpcResponse.ContentType);
-    }
-
-    public static IEnumerable<object[]> DataTests =>
-        new[]{
-            new[]
-            {
-                new byte[] {}
-            },
-            new[]
-            {
-                new byte[] { 0x01, 0x02, 0x03 }
-            }
-        };
-
-    [Theory]
-    [MemberData(nameof(DataTests))]
-    public void ToGetResponseDataTests(byte[] data)
+    [Fact]
+    public void ToGetResponseTests()
     {
-        var response = new StateStoreGetResponse
-        {
-            Data = data
-        };
-
-        var grpcResponse = StateStoreGetResponse.ToGetResponse(response);
-
-        Assert.Equal(data, grpcResponse.Data);
+        var converter = StateStoreGetResponse.ToGetResponse;
+
+        ConversionAssert.ContentTypeEqual(
+            contentType => new StateStoreGetResponse { ContentType = contentType },
+            converter,
+            response => response.ContentType);
+
+        ConversionAssert.DataEqual(
+            data => new StateStoreGetResponse { Data = data },
+            converter,
+            response => response.Data);
+
+        ConversionAssert.ETagEqual(
+            etag => new StateStoreGetResponse { ETag = etag },
+            converter,
+            response => response.Etag);
+
+        ConversionAssert.MetadataEqual(
+            metadata => new StateStoreGetResponse { Metadata = metadata },
+            converter,
+            response => response.Metadata);
```

Comment:
nice

Reply:
I originally thought it'd end up being much more tedious, but it works reasonably well; I'll likely (eventually) refactor some of the previous rounds of tests to use this model.  The main issues is that conversion isn't always symmetric (e.g. .NET SDK uses `string?` but the generated `.proto` uses `string`) which requires some duplication.